src/components/Borrow/WrappedBorrow/WrappedBorrow.tsx

```diff
@@ -0,0 +1,170 @@
+import { printBN } from '@consts/utils'
+import { Grid } from '@material-ui/core'
+import { ExchangeCollateralTokens, ExchangeSyntheticTokens } from '@selectors/solanaWallet'
+import { PublicKey } from '@solana/web3.js'
+import { Vault } from '@synthetify/sdk/lib/exchange'
+import React from 'react'
+import { BorrowInfo } from '../BorrowInfo/BorrowInfo'
+import { BorrowTable } from '../BorrowTable/BorrowTable'
+import { ActionBorrow } from '../SwitchBorrow/ActionBorrow'
+import ActionMenuBorrow, { IActionContents } from '../SwitchBorrow/ActionMenuBorrow'
+import useStyles from './style'
+export interface BorrowedPair extends Vault {
+  collateralData: ExchangeCollateralTokens
+  syntheticData: ExchangeSyntheticTokens
+}
+export interface OwnedVaults {
+  borrowed: string
+  collateral: string
+  deposited: number
+  depositedSign: string
+  cRatio: string
+  currentDebt: number
+  currentDebtSign: string
+  maxBorrow: string
+  interestRate: string
+  liquidationPrice: string
+}
+interface IProp {
+  pairs: BorrowedPair[]
+  ownedVaults: OwnedVaults[]
+  sending: boolean
+  hasError: boolean
+  limit: number
+  reserve: number
+  debtAmount: number
+  collateralAmount: number
+}
+export const WrappedBorrow: React.FC<IProp> = ({
+  pairs,
+  sending,
+  hasError,
+  ownedVaults,
+  limit,
+  reserve,
+  collateralAmount,
+  debtAmount
+}) => {
+  const classes = useStyles()
+  const [cRatio, setCRatio] = React.useState(100.0)
+  const [interestRate, setInterestRate] = React.useState(1)
+  const [liquidationPriceTo, setLiquidationPriceTo] = React.useState(1)
+  const [liquidationPriceFrom, setLiquidationPriceFrom] = React.useState(1)
+
+  const changeCRatio = (nr: number) => {
+    setCRatioTo(cRatio / 10)
+    setCRatio(nr)
+  }
+  const [pairIndex, setPairIndex] = React.useState<number | null>(pairs.length ? 0 : null)
+  const [cRatioTo, setCRatioTo] = React.useState(cRatio / 10)
+  const minCRatio =
+    pairIndex !== null
+      ? Math.pow(
+          Number(
+            printBN(pairs[pairIndex].collateralRatio.val, pairs[pairIndex].collateralRatio.scale)
+          ) / 100,
+          -1
+        )
+      : 0
+
+  const changeValueFromTable = (cRatio: number, interestRate: number, liquidationPrice: number) => {
+    setCRatio(cRatio)
+    setInterestRate(interestRate)
+    setLiquidationPriceTo(liquidationPrice)
+    setCRatioTo(cRatio / 10)
+  }
+
+  const actionOnSubmit = (action: string) => {
```

Comment:
maybe there should be used some callbacks from props, action on submit might be a dispatch of Redux action